hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/s3/security/S3GetSecretRequest.java

```diff
@@ -102,16 +92,42 @@ public OMRequest preExecute(OzoneManager ozoneManager) throws IOException {
     // client does not need any proto changes.
     OMRequest.Builder omRequest = OMRequest.newBuilder()
         .setUserInfo(getUserInfo())
-        .setUpdateGetS3SecretRequest(updateGetS3SecretRequest)
         .setCmdType(getOmRequest().getCmdType())
         .setClientId(getOmRequest().getClientId());
 
+    // createIfNotExist defaults to true if not specified.
+    boolean createIfNotExist = !s3GetSecretRequest.hasCreateIfNotExist()
+            || s3GetSecretRequest.getCreateIfNotExist();
+
+    // Recompose GetS3SecretRequest just in case createIfNotExist is missing
+    final GetS3SecretRequest newGetS3SecretRequest =
+            GetS3SecretRequest.newBuilder()
+                    .setKerberosID(accessId)
```

Comment:
this is confusing since it says setKerberosID() but passing it accessID.

Reply:
With the introduction of multi-tenancy the meaning of the proto field has changed.

Before multi-tenancy, the accessId field is the same as Kerberos ID (and that's probably why the field was named `kerberosID`).

Now we are allowing arbitrary accessIds in multi-tenancy so `accessId == kerberosID` doesn't apply anymore. And the proto message field should have been renamed to `accessId` in previous MT patches.

Reply:
It turns out renaming the proto field from `kerberosID` to `` is failing the proto compatibility check locally.

```
[INFO] --- proto-backwards-compatibility:1.0.5:backwards-compatibility-check (default) @ ozone-interface-client ---
[ERROR] CONFLICT: "GetS3SecretRequest" field: "accessId" ID: 1 has an updated name, previously "kerberosID" [OmClientProtocol.proto]
[ERROR] CONFLICT: "GetS3SecretRequest" field: "kerberosID" has been removed, but is not reserved [OmClientProtocol.proto]
```

I will add a comment in the code without renaming the proto field for the moment.

If we want to do the rename, we should probably do this later in a refactoring patch in the master branch. Because there are a lot of other references of `.setKerberosID`, `.getKerberosID` and in other proto messages as well (not just for `GetS3SecretRequest`).